src/DocumentWatcher.ts

```diff
@@ -172,6 +173,22 @@ function applyEditorConfigToTextEditor(
 	/* tslint:disable:no-any */
 	textEditor.options = <any> newOptions;
 	/* tslint:enable */
+
+	applyEndOfLineSetting();
+
+	function applyEndOfLineSetting() {
+		for (const [regex, eol] of new Map([
+			[/^lf$/i, EndOfLine.LF],
+			[/^crlf$/i, EndOfLine.CRLF]
+		])) {
+			if (regex.test(editorconfig.end_of_line)) {
+				textEditor.edit(edit => {
+					edit.setEndOfLine(eol);
+				});
+				return;
+			}
+		}
```

Comment:
I considered using a hash like that, but then I would only support lowercase `lf` and `crlf` settings. That's why I'm using regex to ensure the case doesn't matter.

I'm not sure yet if this is a transformation or not. I sure hope not. In other words, when someone opens a file it shouldn't transform all the line endings. It should leave the dirty ones alone and only ensure future line endings are correct. As such, it shouldn't be a transform at all, but a setting. Now, I'm thinking vscode doesn't support this strategy, but it's how it's supposed to work. You don't want people going around the project making all the files dirty. That's not the editorconfig strategy.